src/cwe_checker_lib/src/analysis/forward_interprocedural_fixpoint.rs

```diff
@@ -267,19 +268,207 @@ pub fn create_computation<'a, T: Context<'a>>(
+    let priority_sorted_nodes: Vec<NodeIndex> = create_bottom_up_worklist(problem.get_graph());
+    let generalized_problem = GeneralizedContext::new(problem);
+    super::fixpoint::Computation::from_node_priority_list(
+        generalized_problem,
+        default_value.map(NodeValue::Value),
+        priority_sorted_nodes,
+    )
+}
+
 /// Generate a new computation from the corresponding context and an optional default value for nodes.
-/// Uses the alternate worklist order when computing the fixpoint.
+/// Uses a top down worklist order when computing the fixpoint.
 ///
-/// The alternate worklist order moves nodes with 10 or more incoming edges to the end of the priority queue.
-/// This can improve the convergence speed for these nodes in the fixpoint algorithm.
-/// Use if you encounter convergence problems for nodes with a lot of incoming edges.
-pub fn create_computation_with_alternate_worklist_order<'a, T: Context<'a>>(
+/// The worklist order prefers caller nodes before callee nodes.
+pub fn create_computation_with_top_down_worklist_order<'a, T: Context<'a>>(
     problem: T,
     default_value: Option<T::Value>,
 ) -> super::fixpoint::Computation<GeneralizedContext<'a, T>> {
+    let priority_sorted_nodes: Vec<NodeIndex> = create_top_down_worklist(problem.get_graph());
     let generalized_problem = GeneralizedContext::new(problem);
-    super::fixpoint::Computation::new_with_alternate_worklist_order(
+    super::fixpoint::Computation::from_node_priority_list(
         generalized_problem,
         default_value.map(NodeValue::Value),
+        priority_sorted_nodes,
     )
 }
+
+#[cfg(test)]
+mod tests {
+
+    use crate::{
+        analysis::{
+            expression_propagation::Context,
+            forward_interprocedural_fixpoint::create_computation_with_bottom_up_worklist_order,
+        },
+        intermediate_representation::*,
+    };
+    use std::collections::HashMap;
+
+    /// Creates a project with one caller function of two blocks and one callee function of one block.
+    fn mock_project() -> Project {
+        let callee_block = Term {
+            tid: Tid::new("callee_block"),
+            term: Blk {
+                defs: vec![],
+                jmps: vec![Term {
+                    tid: Tid::new("ret"),
+                    term: Jmp::Return(Expression::const_from_i32(42)),
+                }],
+                indirect_jmp_targets: Vec::new(),
+            },
+        };
+        let called_function = Term {
+            tid: Tid::new("called_function"),
+            term: Sub {
+                name: "called_function".to_string(),
+                blocks: vec![callee_block],
+                calling_convention: Some("_stdcall".to_string()),
+            },
+        };
+
+        let caller_block_2 = Term {
+            tid: Tid::new("caller_block_2"),
+            term: Blk {
+                defs: vec![],
+                jmps: vec![],
+                indirect_jmp_targets: Vec::new(),
+            },
```

Comment:
You could write a helper function `new_block(name: &str) -> Term<Blk>` that creates such an empty block. If you replace all three block creations in this function with such a helper function you could save one or 2 lines.